sdk/tests/core/test_az_mqtt5_rpc.c

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "az_test_definitions.h"
+#include <azure/core/az_mqtt5_config.h>
+#include <azure/core/az_mqtt5_rpc.h>
+#include <azure/core/az_result.h>
+#include <azure/core/internal/az_mqtt5_rpc_internal.h>
+
+#include <setjmp.h>
+#include <stdarg.h>
+#include <stddef.h>
+
+#include <cmocka.h>
+
+#define TEST_SAMPLE_FORMAT                                           \
+  "test/" AZ_MQTT5_RPC_CLIENT_ID_KEY "/" AZ_MQTT5_RPC_SERVICE_ID_KEY \
+  "/" AZ_MQTT5_RPC_EXECUTOR_ID_KEY "/" AZ_MQTT5_RPC_COMMAND_ID_KEY "/endoftest"
+#define TEST_SAMPLE_FORMAT_REPLACED                                                    \
+  "$share/test_service_group_id/test/test_client_id/test_service_id/test_executor_id/" \
+  "test_command_id/endoftest\0"
+static const az_span test_sample_format = AZ_SPAN_LITERAL_FROM_STR(TEST_SAMPLE_FORMAT);
+static const az_span test_service_group_id = AZ_SPAN_LITERAL_FROM_STR("test_service_group_id");
+static const az_span test_sample_client_id = AZ_SPAN_LITERAL_FROM_STR("test_client_id");
+static const az_span test_sample_service_id = AZ_SPAN_LITERAL_FROM_STR("test_service_id");
+static const az_span test_sample_executor_id = AZ_SPAN_LITERAL_FROM_STR("test_executor_id");
+static const az_span test_sample_command_id = AZ_SPAN_LITERAL_FROM_STR("test_command_id");
+
+// Calculated hashes shown below. UPDATE whenever tokens change.
+static const uint32_t az_mqtt5_rpc_client_id_hash = 3426466449;
+static const uint32_t az_mqtt5_rpc_service_id_hash = 4175641829;
```

Comment:
These should be removed - we must define the hashes in a single place and verify that the prod code produces the same hashes.